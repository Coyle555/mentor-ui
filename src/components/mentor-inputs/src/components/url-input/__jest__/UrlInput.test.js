import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitForElement, cleanup, act, getByTestId } from '@testing-library/react';

import UrlInput from '../urlInput';

afterEach(cleanup);

test('<UrlInput /> with no props', () => {

	const component = renderer.create(<UrlInput />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});


test('<UrlInput /> with a required attribute', () => {
	const component = renderer.create(
		<UrlInput required />
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('<UrlInput /> with a non-string value passed in', () => {
	const component = renderer.create(
		<UrlInput value={4593.3} />
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('<UrlInput /> with a non-url value passed in', () => {
	const component = renderer.create(
		<UrlInput value="www.goog" />
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('<UrlInput /> with a value passed in', () => {
	const component = renderer.create(
		<UrlInput value="https://www.gmail.com" />
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('<UrlInput /> with autocomplete enabled', () => {
	const component = renderer.create(
		<UrlInput autoComplete="true" />
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
