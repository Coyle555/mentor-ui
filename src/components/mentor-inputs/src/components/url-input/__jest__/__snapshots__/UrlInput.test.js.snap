// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UrlInput /> with a non-string value passed in 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field mui-mi-input-field-has-error"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter URL"
  type="text"
  value="4593.3"
/>
`;

exports[`<UrlInput /> with a non-url value passed in 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field mui-mi-input-field-has-error"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter URL"
  type="text"
  value="www.goog"
/>
`;

exports[`<UrlInput /> with a required attribute 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field mui-mi-input-field-has-error"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter URL"
  required={true}
  type="text"
  value=""
/>
`;

exports[`<UrlInput /> with a value passed in 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter URL"
  type="text"
  value="https://www.gmail.com"
/>
`;

exports[`<UrlInput /> with autocomplete enabled 1`] = `
<input
  autoComplete="true"
  className="mui-mi-input-field"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter URL"
  type="text"
  value=""
/>
`;

exports[`<UrlInput /> with no props 1`] = `
<input
  autoComplete="false"
  className="mui-mi-input-field"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter URL"
  type="text"
  value=""
/>
`;
